lib/evaluation/handlers/uriEvaluationHandler.ts

```diff
@@ -18,17 +20,18 @@ export class UriEvaluationHandler extends AbstractEvaluationHandler {
     return 'UriEvaluation';
   }
 
-  public handle(d: PresentationDefinition, vcs: VerifiableCredential[]): void {
-    d.input_descriptors.forEach((inDesc: InputDescriptor, i: number) => {
-      const uris: string[] = inDesc.schema.map((so) => so.uri);
-      vcs.forEach((vc: VerifiableCredential, j: number) => {
-        this.evaluateUris(UriEvaluationHandler.getPresentationURI(vc), uris, i, j);
+  public handle(d: InternalPresentationDefinition, vcs: InternalVerifiableCredential[]): void {
+    // This filter is removed in V2
+    (<InternalPresentationDefinitionV1>d).input_descriptors.forEach((inDesc: InputDescriptorV1, i: number) => {
+      const uris: string[] = d.getVersion() !== PEVersion.v2 ? inDesc.schema.map((so) => so.uri) : [];
```

Comment:
Not for now, but I guess we could create an enhancement ticket to have evaluationHandlers express their spec version support. Sometimes you have to intervene in the logic during evaluation, but at other times a handler might have to do nothing for a specific spec version. It would be nice if we handled that internally by having a template method around the `handle`, which simply looks at whether it should be executed or not